android/src/main/java/com/reactnativecommunity/netinfo/ConnectivityReceiver.java

```diff
@@ -154,8 +175,33 @@ private WritableMap createConnectivityEventMap() {
                     ConnectivityManagerCompat.isActiveNetworkMetered(getConnectivityManager());
             details.putBoolean("isConnectionExpensive", isConnectionExpensive);
 
+            // Get the cell carrier
+            String carrier = mTelephonyManager.getNetworkOperatorName();
+
             if (mConnectionType.equals(CONNECTION_TYPE_CELLULAR)) {
                 details.putString("cellularGeneration", mCellularGeneration);
+                details.putString("carrier", carrier);
+            }
+
+            // Get the SSID and strip the quotes
+            String initialSSID = mWifiManager.getConnectionInfo().getSSID();
+            String ssid = initialSSID.replace("\"", "");
+
+            // Get/parse the wifi signal strength
+            int initialSignalStrength = mWifiManager.getConnectionInfo().getRssi();
+            String signalStrength = formatSignalStrength(initialSignalStrength);
```

Comment:
I think the API might be more flexible if it just returned numbers to the JS, and let the app format the values the way it wants. According to the Android docs, you can use `WifiManager.calculateSignalLevel()` to convert the signal strength into a range like `0-5`. Then the app can format as English strings or something else (e.g. icons which indicate strength).
https://developer.android.com/reference/android/net/wifi/WifiManager#calculateSignalLevel(int,%20int)